tests/perf/benchmark.py

```diff
@@ -0,0 +1,200 @@
+"""OTX Benchmark based on tools/experiment.py."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import os
+import glob
+import pandas as pd
+import yaml
+from pathlib import Path
+from typing import List, Optional
+
+from tests.test_suite.run_test_command import check_run
+
+
+class OTXBenchmark:
+    """Benchmark runner based on tools/experiment.py in OTX1.x.
+
+    Example:
+        >>> bm = OTXBenchmark(['random_sample1', 'random_sample2'], data_root='./data/coco')
+        >>> atss_result = bm.run('MobileNetV2-ATSS')
+        >>> yolox_result = bm.run('YOLOX-TINY')
+
+    Args:
+        datasets (List[str]): Paths to datasets relative to the data_root.
+            Intended for, but not restricted to different sampling based on same dataset.
+        data_root (str): Path to the root of dataset directories. Defaults to './data'.
+        num_epoch (int): Overrides the per-model default number of epoch settings.
+            Defaults to 0, which means no overriding.
+        num_repeat (int): Number for trials with different random seed, which would be set
+            as range(0, num_repeat). Defaults to 1.
+        train_params (dict): Additional training parameters.
+            e.x) {'learning_parameters.num_iters': 2}. Defaults to {}.
+        track_resources (bool): Whether to track CPU & GPU usage metrics. Defaults to False.
+        eval_upto (str): The last serial operation to evaluate. Choose on of ('train', 'export', 'optimize').
+            Operations include the preceeding ones.
+            e.x) Eval up to 'optimize': train -> eval -> export -> eval -> optimize -> eval
+            Default to 'train'.
+        output_root (str): Output root dirctory for logs and results. Defaults to './otx-benchmark'.
+        dry_run (bool): Whether to just print the OTX command without execution. Defaults to False.
+        tags (dict): Key-values pair metadata for the experiment. Defaults to {}.
+    """
+
+    def __init__(
+        self,
+        datasets: List[str],
+        data_root: str = "data",
+        num_epoch: int = 0,
+        num_repeat: int = 1,
+        train_params: dict = {},
+        track_resources: bool = False,
+        eval_upto: str = "train",
+        output_root: str = "otx-benchmark",
+        dry_run: bool = False,
+        tags: dict = {},
+        **kwargs,
+    ):
+        self.datasets = datasets
+        self.data_root = data_root
+        self.num_epoch = num_epoch
+        self.num_repeat = num_repeat
+        self.train_params = train_params
+        self.track_resources = track_resources
+        self.eval_upto = eval_upto
+        self.output_root = output_root
+        self.dry_run = dry_run
+        self.tags = tags
+        self.subset_dir_names = kwargs.get("subset_dir_names", {"train": "", "val": "", "test": ""})
+
+    def run(
+        self,
+        model_id: str,
+        train_params: dict = {},
+        tags: dict = {},
+    ) -> pd.DataFrame:
```

Comment:
```suggestion
    ) -> pd.DataFrame | None:
```
None can be returned in some cases, I think.

Reply:
Fixed in https://github.com/openvinotoolkit/training_extensions/pull/2742/commits/91bfec13541e70258511473e29b8b56dfed4c3d1